src/main/kotlin/com/swm_standard/phote/service/ExamService.kt

```diff
@@ -0,0 +1,41 @@
+package com.swm_standard.phote.service
+
+import com.swm_standard.phote.common.exception.NotFoundException
+import com.swm_standard.phote.dto.ReadExamHistoryDetail
+import com.swm_standard.phote.dto.ReadExamHistoryDetailResponse
+import com.swm_standard.phote.repository.ExamRepository
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+
+@Service
+class ExamService(
+    private val examRepository: ExamRepository
+) {
+    @Transactional(readOnly = true)
+    fun readExamHistoryDetail(id: Long): ReadExamHistoryDetailResponse {
+        val exam = examRepository.findById(id).orElseThrow { NotFoundException("examId", "존재하지 않는 examId") }
+        val responses = mutableListOf<ReadExamHistoryDetail>()
+
+        exam.answers.map { answer ->
+            responses.add(
+                ReadExamHistoryDetail(
+                    statement = answer.question.statement,
+                    options = answer.question.options?.let { answer.question.deserializeOptions() },
+                    image = answer.question.image,
+                    category = answer.question.category,
+                    answer = answer.question.answer,
+                    submittedAnswer = answer.submittedAnswer,
+                    isCorrect = answer.isCorrect,
+                    sequence = answer.sequence
+                )
+            )
+        }
+
```

Comment:
NIT) buildList 를 사용해서 선언과 동시에 초기화 하면 mutableList를 쓰지 않아서 더 좋을거같아여~! 혹시 반영하실거라면, 로컬에서 테스트 해보시길 추천드립니다!

```suggestion
        val responses = buildList {
        exam.answers.forEach { answer ->
        add(
            ReadExamHistoryDetail(
                statement = answer.question.statement,
                options = answer.question.options?.let { answer.question.deserializeOptions() },
                image = answer.question.image,
                category = answer.question.category,
                answer = answer.question.answer,
                submittedAnswer = answer.submittedAnswer,
                isCorrect = answer.isCorrect,
                sequence = answer.sequence
            )
        )
    }
}

```

Reply:
로컬에서 확인해보고 적용해보겠습니다~~~~ 야호~~ 현직자 코리 귀하다 .. . . . .